core/quantity/src/test/Formatting.test.ts

```diff
@@ -815,4 +815,122 @@ describe("Formatting tests:", () => {
     assert.isTrue(Format.isFormatTraitSetInProps(formatProps, FormatTraits.ZeroEmpty));
   });
 
+  it("deserialization of custom Formats is consistent", async () => {
+    const unitsProvider = new TestUnitsProvider();
+    const formatProps: CustomFormatProps = {
+      formatTraits: ["KeepSingleZero", "ApplyRounding", "KeepDecimalPoint", "ShowUnitLabel", "Use1000Separator"],
+      precision: 6,
+      type: "Decimal",
+      uomSeparator: " ",
+      custom: {
+        foo: "bar",
+      },
+    };
+
+    const format = await Format.createFromJSON("test", unitsProvider, formatProps);
+
+    const jsonObj: FormatProps = format.toJSON();
+
+    // Assertions
+    // eslint-disable-next-line guard-for-in
+    for (const key in formatProps) {
```

Comment:
I believe you may be able to use chai's assert.deepEqual here

Reply:
I originally used that as well, but it can't be done because of `formatTraits`. deepEqual expects the order of two compared arrays to match too, which is something we reasonably don't care about. That's why I wrote down logic within for comparing members of the 2 arrays, without caring about the order